### PR TITLE
[CRON] implement basic scheduler job for running task

### LINKED ISSUE #14: [CRON] implement basic scheduler job for running task
*Biz Requirements*

- create a basic CRON scheduler that would run every 5 min
- CRON expression should be configurable and easy to change (e.g. move to application properties)
- CRON should call some method generic method `checkHostsAvailability()` in `MonitorService.class`
- add tests that CRON was triggered properly

*Documentation*
- https://www.baeldung.com/spring-scheduled-tasks

### PR DESCRIPTION
**Pull request by AntonTsa**

### CODE REVIEW

**Review by hrafov — COMMENTED** [review]
maybe we need to delete all comments like //@Scheduled(cron = "${scheduled.cron}")
and maybe we need to use a properties file to externalize the configuration.

**Comment by AntonTsa** [comment]
> maybe we need to delete all comments like //@scheduled(cron = "${scheduled.cron}") and maybe we need to use a properties file to externalize the configuration.

I tried to do it using properties, but when I tested it, I got:
image

File: src/test/java/com/example/statussvc/service/MonitorServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Test
+    @DisplayName("""
+            GIVEN valid scheduled task
+            WHEN await at most 10 minutes
+            THEN verify trigger task at least twice
+            """)
+    public void properlyTrigger() {
+        await()
+                .atMost(Duration.ofMinutes(10L))
```
Comment (yurii-polishchuk) [inline src/test/java/com/example/statussvc/service/MonitorServiceTest.java]: not a good idea to have test that just waits 10min

File: src/main/java/com/example/statussvc/service/MonitorService.java [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+@EnableScheduling
+@ConditionalOnProperty(name = "scheduler.enabled", matchIfMissing = true)
 public class MonitorService {
 
     /**
      * Add docs
      */
+
+    //@Scheduled(cron = "${scheduled.cron}")
+    @Scheduled(cron = "0 */5 * * * *")
```
Comment (yurii-polishchuk) [inline src/main/java/com/example/statussvc/service/MonitorService.java]: just  
```
@Scheduled(cron = "${application.scheduler.host.availability.cron}")
```

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -58,3 +58,7 @@ logging:
   file:
     name: ${spring.application.name}.log
   config: classpath:log4j2.yml
+
+#Scheduling
```
Comment (yurii-polishchuk) [inline src/main/resources/application.yml:66]: please move under applciation e.g.

`application.scheduler.host.availability.cron: "0 */5 * * * *"

File: src/main/java/com/example/statussvc/service/MonitorService.java [hunk 4]
```diff
@@ -1,16 +1,26 @@
...
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
 
 /**
  * Generic Monitor Service
  */
 @Log4j2
+@Service
+@EnableScheduling
+@ConditionalOnProperty(name = "scheduler.enabled", matchIfMissing = true)
```
Comment (yurii-polishchuk) [inline src/main/java/com/example/statussvc/service/MonitorService.java]: why do we need conditions?

**Review by yurii-polishchuk — CHANGES_REQUESTED** [review]

File: src/test/java/com/example/statussvc/service/MonitorServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import java.time.Duration;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+@SpringJUnitConfig(MonitorService.class)
+@ActiveProfiles("test")
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class MonitorServiceTest {
```
Comment (yurii-polishchuk) [inline src/test/java/com/example/statussvc/service/MonitorServiceTest.java:24]: @EnableConfigurationProperties(ApplicationProperties.class)
 - https://www.baeldung.com/spring-enable-config-properties

**Review by yurii-polishchuk — CHANGES_REQUESTED** [review]

File: src/test/java/com/example/statussvc/service/MonitorServiceTest.java (hunk 5 above)
Comment (yurii-polishchuk) [reply src/test/java/com/example/statussvc/service/MonitorServiceTest.java:24]: @TestPropertySource(properties = "application.scheduler.host.availability.cron: 0 */5 * * * *")
 - https://www.baeldung.com/spring-test-property-source

**Review by yurii-polishchuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**